### PR TITLE
fix(gui-v2): email me in form view

### PR DESCRIPTION
**Pull request by wingkwong**
## Change Summary

- frontend part : ref: #3357
- backend part: fix broken logic in v1

## Change type

- [ ] feat: (new feature for the user, not a new feature for build script)
- [x] fix: (bug fix for the user, not a fix to a build script)
- [ ] docs: (changes to the documentation)
- [ ] style: (formatting, missing semi colons, etc; no production code change)
- [ ] refactor: (refactoring production code, eg. renaming a variable)
- [ ] test: (adding missing tests, refactoring tests; no production code change)
- [ ] chore: (updating grunt tasks etc; no production code change)

## Test/ Verification

image

### CODE REVIEW

**Review by dstala — CHANGES_REQUESTED** [review]
@wingkwong `email me` option works even after we reset SMTP configuration. should work only when we have an active SMTP configuration.

**Review by pranavxc — APPROVED** [review]

File: packages/nocodb/src/lib/meta/helpers/webhookHelpers.ts [hunk 1]
```diff
@@ -268,16 +267,11 @@ export function _transformSubmittedFormDataForEmail(
   // @ts-ignore
   formView,
   // @ts-ignore
-  columns: Column[]
+  columns: Record<string, any>[]
```
Comment (pranavxc) [inline packages/nocodb/src/lib/meta/helpers/webhookHelpers.ts:270]: The type could be ColumnType.

Comment (wingkwong) [reply packages/nocodb/src/lib/meta/helpers/webhookHelpers.ts:270]: `columns` includes `Column[]` and some extra fields.`ColumnType` shouldn't be used as well.

Comment (pranavxc) [reply packages/nocodb/src/lib/meta/helpers/webhookHelpers.ts:270]: Then you can do `(Column &  { additionalTypes : Type })[]`

**Review by dstala — CHANGES_REQUESTED** [review]
broken. unable to enable `email me` option after configuring SMTP
I tried over SQLite xcdb project

**Comment by wingkwong** [comment]
@dstala Got the cache problem. Should be good now.

image

**Review by dstala — APPROVED** [review]

**Pull request merged**